### PR TITLE
[Perf] Add new npu_fused_infer_attention_score op to improve perfomance in splitfuse cases and resolve long-seq mask problems

### PR DESCRIPTION
**Pull request by qyqc731**
### What this PR does / why we need it?
Add new npu_fused_infer_attention_score op to improve perfomance in splitfuse cases and resolve long-seq mask problems .

1. The original op's performance is suboptimal in certain scenarios, necessitating optimization through the _new op_ (npu_fused_infer_attention_score)。
2. For ultra-long sequences (128k), the original operator will allocate a large attn_mask, which consumes excessive CPU memory. In contrast, the _new op_ supports a fixed-size compressed mask, effectively resolving this issue.

NOTE1: The current PR retains the original logic and uses a version check of the CANN package to determine whether the _new op_ can be enabled. This ensures no impact on existing users. In future versions, this version check and the original logic will be deprecated, and the _new op_ scheduling will be uniformly adopted.
NOTE2: This pr relies on future CANN version, which is not available now.
NOTE3: To enable the new op in chunked prefill, the parameter additional_config should be set like `--additional-config '{"ascend_scheduler_config": {"enabled":true,"enable_chunked_prefill":true}}' \` at least.

### Does this PR introduce _any_ user-facing change?
No
### How was this patch tested?
CI passed

- vLLM version: v0.10.2
- vLLM main: https://github.com/vllm-project/vllm/commit/6c5f82e5aa87cd73ce03ce10fc44138f75ee1aea

### CODE REVIEW

**Comment by rjg-lyh** [comment]
Could you integrate other scenarios, such as full FlashAttention, using the FIA interface as well, and provide the performance test results?

File: vllm_ascend/worker/model_runner_v1.py [hunk 1]
```diff
@@ -293,7 +293,7 @@ def __init__(self, vllm_config: VllmConfig, device: torch.device):
         )
 
         self.attn_mask_builder = AttentionMaskBuilder(
-            self.model_config.max_model_len, self.dtype)
+            self.model_config.max_model_len, self.dtype, self.device)
```
Comment (Angazenn) [inline vllm_ascend/worker/model_runner_v1.py]: If we pass `self.model_config.max_model_len` here we still have an attention mask of shape [max_model_len, max_model_len]. We may have 2 choices:
1. Passing `self.scheduler_config.max_num_batched_tokens` instead of `self.model_config.max_model_len`.
2. Also using new attention op for full prefill case.

File: vllm_ascend/attention/attention_v1.py [hunk 2]
```diff
@@ -455,18 +455,25 @@ def _forward_v1_style(
...
+            key=key,
+            value=value,
+            atten_mask=attn_metadata.attn_mask.to(device=query.device),
             block_table=attn_metadata.block_tables,
-            seq_len=attn_metadata.query_lens,
-            context_lens=attn_metadata.seq_lens,
-            num_kv_heads=self.num_kv_heads,
+            input_layout="TND",
+            block_size=block_size,
+            actual_seq_lengths=attn_metadata.query_start_loc[1:],
```
[the remark this answers is not in the document]
Comment (Angazenn) [reply vllm_ascend/attention/attention_v1.py]: plz consider use `attn_metadata.query_lens`

File: vllm_ascend/worker/model_runner_v1.py (hunk 1 above)
Comment (Angazenn) [inline vllm_ascend/worker/model_runner_v1.py]: for choice 1:
```suggestion
            self.scheduler_config.max_num_batched_tokens, self.dtype, self.device)
```

**Comment by Angazenn** [comment]
Also notice another 2 points:
1. compatibility with difference torch_npu/cann version.
2. fix bug when padding is introduced in dp (len of query exceeds attn_metadata.query_lens[-1]).

File: vllm_ascend/worker/model_runner_v1.py [hunk 3]
```diff
@@ -797,8 +797,7 @@ def _make_attention_mask(self, seq_lens, position,
                              attn_state) -> torch.Tensor:
         # Chunk Prefill situation.
         if attn_state == AscendAttentionState.ChunkedPrefill and not self.vllm_config.model_config.use_mla:
-            return self.attn_mask_builder.get_splitfuse_attn_mask(
-                seq_lens, position, self.dtype, self.device)
+            return torch.triu(torch.ones(2048, 2048), diagonal=1).to(torch.int8)
```
[the remark this answers is not in the document]
Comment (Angazenn) [reply vllm_ascend/worker/model_runner_v1.py]: this is because new op support compressed mask

File: vllm_ascend/attention/attention_v1.py [hunk 4]
```diff
@@ -455,18 +458,39 @@ def _forward_v1_style(
...
-            value_cache=self.value_cache,
-            mask=attn_metadata.attn_mask,
-            block_table=attn_metadata.block_tables,
-            seq_len=attn_metadata.query_lens,
-            context_lens=attn_metadata.seq_lens,
-            num_kv_heads=self.num_kv_heads,
-            num_heads=self.num_heads,
-            scale_value=self.scale,
-            out=output)
+        if self.compressed_mask:
```
Comment (qyqc731) [inline vllm_ascend/attention/attention_v1.py]: add TODO: This op will be used in more situation in future.

File: vllm_ascend/worker/model_runner_v1.py [hunk 5]
```diff
@@ -797,8 +804,11 @@ def _make_attention_mask(self, seq_lens, position,
                              attn_state) -> torch.Tensor:
         # Chunk Prefill situation.
         if attn_state == AscendAttentionState.ChunkedPrefill and not self.vllm_config.model_config.use_mla:
-            return self.attn_mask_builder.get_splitfuse_attn_mask(
-                seq_lens, position, self.dtype, self.device)
+            if selkf.compressed_mas:
+                return self.attn_mask_builder.get_splitfuse_attn_mask()
```
Comment (weijinqian0) [inline vllm_ascend/worker/model_runner_v1.py:827]: code bug

File: vllm_ascend/utils.py [hunk 6]
```diff
@@ -617,3 +617,28 @@ def weak_ref_tensors(
     if isinstance(tensors, tuple):
         return tuple(weak_ref_tensor(t) for t in tensors)
     raise ValueError("Invalid type for tensors")
+
+
+def verify_torch_npu_version(
```
Comment (qyqc731) [inline vllm_ascend/utils.py]: The cann version needs to be verfied. using package.version

Comment (qyqc731) [reply vllm_ascend/utils.py]: add todo for temp

File: vllm_ascend/attention/attention_mask.py [hunk 7]
```diff
@@ -39,11 +39,19 @@ def __init__(
...
 
         self._seq_len_cached = attn_mask.shape[0]
         self.attn_mask_cache = attn_mask
+        self.device = device
+        if self.device:
+            #NOTE: New compressed mask needs to be sent to certain device, 
+            # so device needs to be passed here.
+            assigned_mask_dim = 2048
+            self.chunked_prefill_attn_mask = torch.triu(torch.ones(assigned_mask_dim, assigned_mask_dim), diagonal=1
+            ).to(torch.int8).to(device)
```
Comment (qyqc731) [inline vllm_ascend/attention/attention_mask.py]: check type again

File: vllm_ascend/attention/attention_mask.py [hunk 8]
```diff
@@ -39,11 +39,19 @@ def __init__(
         self,
         max_seq_len: int,
         dtype: torch.dtype,
+        device: torch.device = None,
```
Comment (MengqingCao) [inline vllm_ascend/attention/attention_mask.py:42]: Let's make some comments to describe what the `device` mean, and make clear that we should only pass by this arg when using the new attention op

File: vllm_ascend/attention/attention_mask.py (hunk 7 above)
Comment (Angazenn) [reply vllm_ascend/attention/attention_mask.py]: this op currently only supports int8 mask

File: vllm_ascend/worker/model_runner_v1.py (hunk 5 above)
Comment (Angazenn) [reply vllm_ascend/worker/model_runner_v1.py:827]: fixed

File: vllm_ascend/attention/attention_mask.py (hunk 8 above)
Comment (qyqc731) [reply vllm_ascend/attention/attention_mask.py:42]: Added
